viewing_party/party.py

```diff
@@ -1,23 +1,213 @@
+
+    list_of_movie_ratings=[]
+    sum_of_all_ratings=0
+
+    if len(list_of_watched_movie_info) == 0:
+        return 0.0
+
+    #Adds rating of each watched movie to a list
+    for movie_info in list_of_watched_movie_info:
+        list_of_movie_ratings.append(movie_info["rating"])
+
+    #Determines the sum of all ratings
+    for rating in list_of_movie_ratings:
+        sum_of_all_ratings+=rating
+
+    #Calculates the average rating using the sum of all ratings and the number of movies
+    average_rating=sum_of_all_ratings/number_of_watched_movies
+    return average_rating
+
+def get_most_watched_genre(single_user_data):
+    list_of_watched_movie_info=single_user_data["watched"]
+    list_of_genres=[]
+    genre_counter_dict={}
+    list_of_watched_genre_counts=[]
+
+    if len(list_of_watched_movie_info) == 0:
+        return None
+
+    #Adds genre of each watched movie to a list
+    for movie_info in list_of_watched_movie_info:
+        list_of_genres.append(movie_info['genre'])
+
+    #Adds genre as the key to a dictionary with a value that counts the number of times it is in the genre list
+    for genre in list_of_genres:
+        if not genre in genre_counter_dict:
+            genre_counter_dict[genre] = 1
+        else:
+            genre_counter_dict[genre] += 1
+
+    #Adds genre count values to a list
+    for genre_count_value in genre_counter_dict.values():
+        list_of_watched_genre_counts.append(genre_count_value)
+
+    #Determines the highest genre count value
+    max_watched_genre_count=max(list_of_watched_genre_counts)
+
+    #Returns a genre from the genre:counter dictionary with a counter value that is the same as the highest genre count value
+    for genre,genre_count_value in genre_counter_dict.items():
+        if genre_count_value == max_watched_genre_count:
+            return genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(single_user_data):
+    list_of_user_movie_info=single_user_data["watched"]
+    list_of_frnds_watched_dicts=single_user_data["friends"]
+    list_of_user_movie_titles=[]
+    list_of_friends_movie_titles=[]
+    list_of_unique_user_movie_info=[]
+    
+    #Adds user's watched movie titles to a list
+    for user_movie_info in list_of_user_movie_info:
+        list_of_user_movie_titles.append(user_movie_info["title"])
+ 
+    #Adds friends' watched movie titles to a list
+    for frnds_watched_movie_info in list_of_frnds_watched_dicts:
+        for frnds_watched_movie_titles in frnds_watched_movie_info["watched"]:
+            list_of_friends_movie_titles.append(frnds_watched_movie_titles["title"])
+
+    #Uses sets data structure and difference method to add user's unique move titles that only the user has seen to a list
+    list_of_unique_user_movie_titles=set(list_of_user_movie_titles).difference(set(list_of_friends_movie_titles))

```

Comment:
nice job using sets to help filter out duplicates! we could use them immediately on lines 103 and 104 by setting `list_of_friends_movie_titles=()`, then loop over the user's movies

Any time it tries to add a value, it has to check if it's already in there. If it is, it won't add it. 

Your solution works just fine, though! something to simply consider